client/src/components/SignUpModal.jsx

```diff
@@ -0,0 +1,121 @@
+import { useState } from "react"
+import Modal from "react-bootstrap/Modal"
+import Button from "react-bootstrap/Button"
+import Form from 'react-bootstrap/Form'
+import Alert from 'react-bootstrap/Alert'
+import { sendSignUp } from "../lib/api"
+
+/**
+ * Function to create the signup modal and handle a user signing up.
+*/
+export default function SignUpModal({ showSignUp, setShowSignUp, setShowSignIn }) {
+  const [userName, setUserName] = useState('')
+  const [password, setPassword] = useState('')
+  const [verifyPassword, setVerifyPassword] = useState('')
+  const [firstName, setFirstName] = useState('')
+  const [lastName, setLastName] = useState('')
+  const [email, setEmail] = useState('')
+  const [passwordMatch, setPasswordMatch] = useState()
```

Comment:
Does this need to be state or can you just compare password and verifyPassword when you need this value?